ow_power_system/src/power_system_node.cpp

```diff
@@ -17,27 +25,117 @@ int main(int argc, char* argv[]) {
 
   //Construct our power draw listener
   //TODO: Check out latching
-  ros::Subscriber power_listener = n.subscribe("power_draw", 1000, computeSOC);
+  //ros::Subscriber power_listener = n.subscribe("power_draw", 1000, computeSOC);
 
-  //Construct our State of Charge (SOC) publisher
+  //Construct our State of Charge (SOC) publisher (Volts)
   ros::Publisher SOC_pub = n.advertise<std_msgs::Float64>("state_of_charge",1000);
+  //Construct our Remaining Useful Life (RUL) publisher (Seconds)
+  ros::Publisher RUL_pub = n.advertise<std_msgs::Float64>("remaining_useful_life",1000);
+  
+  //Load power values csv
+  string csv_path,csv_file;
+  csv_file = "/data/onewatt.csv";
+  //FILE OPTIONS: onewatt.csv, eightwatt.csv, sixteenwatt.csv
+  csv_path = ros::package::getPath("ow_power_system");
+  csv_path += csv_file;
+
+  auto power_csv = loadCSV(csv_path);
+  
+  //Define our publication rate and initialize our time to 0
+  double power_update_rate;  
+  if (csv_file=="/data/onewatt.csv"){
+    power_update_rate = 0.10; //csv is at 10Hz intervals
+  } else {
+    power_update_rate = 1.0; // 1 Hz default
+  }
+  int t_step= 1;//starting at 1 not zero to skip csv headings
 
+  //set our indices
+  int time_i = 0;
+  int voltage_i = 1;
+  int rul_i = 2;
+  
   // ROS Loop. Note that once this loop starts,
   // this function (and node) is terminated with an interrupt.
-  ros::Rate rate(1); // 1 Hz, tailor as needed
-  while (ros::ok()) {
-    //individual soc_msg to be published by SOC_pub
-    std_msgs::Float64 soc_msg;
+  
+  ros::Rate rate(power_update_rate);
+  //individual soc_msg to be published by SOC_pub
+  std_msgs::Float64 soc_msg;
+  std_msgs::Float64 rul_msg;
+  ROS_INFO ("Power system node running");
 
-    soc_msg.data=0.0; //zero for testing purposes, will be calculated later
+  int power_lifetime = power_csv.size();
+  while (ros::ok()) {
+    if (t_step < power_lifetime) {
+      soc_msg.data=power_csv[t_step][voltage_i];
+      rul_msg.data=power_csv[t_step][rul_i];
+    } //should keep publishing last voltage value if sim time exceeds the length of the csv
+    
+    //soc_msg.data=t;
```

Comment:
This looks like more work to do, so maybe make another Jira ticket.

Reply:
Nope, I did do that! I'll get rid of the comment.